### DISCUSSION TITLE
Some jobs get cancelled in phase 3

### DISCUSSION THREAD

**Discussion by EnderElement** [swar/Swar-Chia-Plot-Manager#411] [Q&A]
> Starting phase 3/4: Compression from tmp files into "" ... Sat May 15 16:53:14 2021
Compressing tables 1 and 2
Caught plotting error: bad allocation
[8024] Failed to execute script chia
Traceback (most recent call last):
  File "chia\cmds\chia.py", line 81, in <module>
  File "chia\cmds\chia.py", line 77, in main
  File "click\core.py", line 829, in __call__
  File "click\core.py", line 782, in main
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1259, in invoke
  File "click\core.py", line 1066, in invoke
  File "click\core.py", line 610, in invoke
  File "click\decorators.py", line 21, in new_func
  File "chia\cmds\plots.py", line 135, in create_cmd
  File "chia\plotting\create_plots.py", line 176, in create_plots
RuntimeError: bad allocation

Settings:
> size: 32
    bitfield: true
    threads: 20
    buckets: 128
    memory_buffer: 32000
    max_concurrent: 3
    max_concurrent_with_start_early: 4
    stagger_minutes: 120
    max_for_phase_1: 3
    concurrency_start_early_phase: 3
    concurrency_start_early_phase_delay: 0
    temporary2_destination_sync: false

Before (note num 3):
![image](https://user-images.githubusercontent.com/29475587/118354475-94c7ef80-b5ae-11eb-80d3-f3aa5f32c340.png)

After:
![image](https://user-images.githubusercontent.com/29475587/118354486-a3160b80-b5ae-11eb-8aea-29610d0c9201.png)

- Plotter1 is a 1TB SSD and Plotter2 is a 512GB SSD
- Chia version 1.1.5
- CPU is a Ryzen 9 5950X
- 4x16GB 3200MHz CL16 RAM

I'm trying to find the best settings for Plotter1 and it theoretically can support four parallel jobs if staggered so I set concurrency_start_early_phase to 3. This only happens if when there are over two parallel jobs, also with other settings and not just the one above, although possibly due to concurrency_start_early_phase being 3. When this error happens to job isn't only cancelled, but the temp files remain so if I leave the manager running for a long time, the plotting SSD will have a build up of garbage and eventually run out of space.

**Comment by EnderElement** [comment]
I was under the asumption that the memory buffer and threads are a total limit for all parallel jobs on one drive because at the heaviest point, memory usage never went above 40%.

**Comment by swar** [comment]
It is not for the entire job, it is for each individual plot process. This is the same value you would enter in the Chia GUI when kicking off a new plot.

Your memory probably don't go above 40% because most of it was page filed.
